src/Umbraco.AuthorizedServices/Services/Implement/AuthorizedServiceCaller.cs

```diff
@@ -111,7 +118,10 @@ public async Task<string> SendRequestRawAsync<TRequest>(string serviceAlias, str
     public string? GetApiKey(string serviceAlias)
     {
         ServiceDetail serviceDetail = GetServiceDetail(serviceAlias);
-        return serviceDetail?.ApiKey;
+        string? key = KeyStorage.GetKey(serviceAlias);
```

Comment:
Again, should check for a configured key first to remove an unnecessary database call if one is configured.